### DISCUSSION TITLE
MongoDB Instances View Internal Server Error

### DISCUSSION THREAD

**Discussion by gustavoaraujocardozo** [elsa-workflows/elsa-core#5707] [Q&A]
I have setting the project to run with mongo db, like this...

	elsa.UseMongoDb($"{mongoDbConnectionString}/{mongoDbName}");

	elsa.UseWorkflowManagement(management =>
	{
		management.UseMongoDb();
	});

	elsa.UseWorkflowRuntime(runtime =>
	{
		runtime.UseMongoDb();
	});

after execute my workflow while checking the instances, i got this error

System.FormatException: An error occurred while deserializing the Outputs property of class Elsa.Workflows.Runtime.Entities.ActivityExecutionRecord: An error occurred while deserializing the MemoryBlockReference property of class Elsa.Workflows.Models.Argument: Type is not supported.
 ---> System.FormatException: An error occurred while deserializing the MemoryBlockReference property of class Elsa.Workflows.Models.Argument: Type is not supported.
 ---> System.ArgumentException: Type is not supported.
   at System.Runtime.CompilerServices.RuntimeHelpers.GetUninitializedObject(QCallTypeHandle type, ObjectHandleOnStack retObject)
   at System.Runtime.CompilerServices.RuntimeHelpers.GetUninitializedObject(Type type)
   at MongoDB.Bson.Serialization.BsonClassMapSerializer`1.DeserializeClass(BsonDeserializationContext context)
   at MongoDB.Bson.Serialization.BsonClassMapSerializer`1.Deserialize(BsonDeserializationContext context, BsonDeserializationArgs args)
   at MongoDB.Bson.Serialization.Serializers.SerializerBase`1.MongoDB.Bson.Serialization.IBsonSerializer.Deserialize(BsonDeserializationContext context, BsonDeserializationArgs args)
   at MongoDB.Bson.Serialization.BsonClassMapSerializer`1.DeserializeMemberValue(BsonDeserializationContext context, BsonMemberMap memberMap)

![image](https://github.com/elsa-workflows/elsa-core/assets/11926577/bcdf5c8f-cc34-4df2-a349-282df7695b74)

I got the error at this point when call this endpoint (https://localhost:7268/elsa/api/activity-executions/report)

![image](https://github.com/elsa-workflows/elsa-core/assets/11926577/def43c9b-ecce-4b91-bf97-2a4d2dd0c118)

the same is not true while using sql lite, just with mongo db. The problem start when i click on executions tab.

![image](https://github.com/elsa-workflows/elsa-core/assets/11926577/0c0da361-5506-478f-bdb7-20d03fe2bd8a)

**Comment by glime-ai** [comment]
It seems like you are encountering a serialization issue with MongoDB when using Elsa Workflows. This type of error typically occurs when the MongoDB driver is unable to deserialize a field in a document because the type of the field is not supported or not recognized by the serializer.

To resolve this issue, you may need to:

1. Ensure that all custom types used in your workflows are properly registered with the MongoDB serializer. You can do this by implementing a custom `BsonClassMap` for your types.
2. Check if there are any known issues or updates related to MongoDB serialization in the Elsa GitHub repository or discussions.
3. Review the MongoDB serialization documentation to ensure that your types are compatible with the MongoDB BSON serializer.

If you are unable to resolve the issue on your own, consider reaching out to the Elsa community or maintainers for assistance. You can create an issue on the Elsa GitHub repository with detailed information about your problem, including the version of Elsa you are using, the MongoDB version, and any relevant code snippets or configuration details.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by gustavoaraujocardozo** [reply]
It didin't help there is a lot of properties in many classes with the same problem, ActivityExecutionRecord, Argument, Output.
